tests/filter-tests.js

```diff
@@ -438,4 +438,42 @@ describe("broccoli-asset-rev", function () {
       `,
     });
   });
+
+  it('can optionally use caching', async function() {
+    this.timeout(500);
+    let fixturesPath = `${__dirname}/fixtures`;
+    let fixtures = fromDir(fixturesPath);
+    const input = await createTempDir();
+    input.write(fixtures.read("js-perf/input"));
+
+    const subject = new AssetRewrite(input.path(), {
+      assetMap: {
+        assetMap: JSON.parse(
+          fs.readFileSync(`${fixturesPath}/js-perf/asset-map.json`)
+        ),
+        replaceExtensions: ["js"],
+        enableCaching: true
+      },
+    });
+    const output = createBuilder(subject);
+
+    const run1Start = process.hrtime();
+    await output.build();
+    let expectedTestSupportOutput = fs.readFileSync(`${fixturesPath}/js-perf/output/test-support.js`).toString();
+    const run1End = process.hrtime(run1Start);
+    expect(output.read()).to.deep.equal({
+      'test-support.js': expectedTestSupportOutput
+    });
+
+    const run2Start = process.hrtime();
+    await output.build();
+    const run2End = process.hrtime(run2Start);
+    expect(output.read()).to.deep.equal({
+      'test-support.js': expectedTestSupportOutput
+    });
+    
+    const run1DurationMs = (run1End[0] * 1000000000 + run1End[1]) / 1000000;
+    const run2DurationMs = (run2End[0] * 1000000000 + run2End[1]) / 1000000;
+    expect(run2DurationMs).to.be.below(run1DurationMs * .75, `cache should make second run (${run2DurationMs}ms) at least 25% faster than first run (${run1DurationMs}ms)`);
```

Comment:
I worry about this in the long run. I've been burned by assumptions around timing semantics (especially in CI) many times.

I think the idea here is that we want to confirm that we are properly avoiding extra work. Could we do the same by exposing some counters on the `AssetRewrite` instance?

Reply:
See what you think about the change I made here.